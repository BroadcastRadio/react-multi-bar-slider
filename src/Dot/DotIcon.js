import PropTypes from 'prop-types';
import glamorous from 'glamorous';

const DotIcon = glamorous.img({
  position: 'absolute',
  transform: 'translateX(-50%)',
  userDrag: 'none',
  userSelect: 'none'
}, ({ width, height }) => ({
  width,
  height
}));

DotIcon.displayName = 'DotIcon';

DotIcon.propTypes = {
  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  css: PropTypes.object
};

DotIcon.defaultProps = {
  width: 50,
  height: 50
};

export default DotIcon;
