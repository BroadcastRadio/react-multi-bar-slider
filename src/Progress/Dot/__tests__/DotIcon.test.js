import React from 'react';
import Enzyme, { render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import DotIcon from '../DotIcon';

Enzyme.configure({ adapter: new Adapter() });

beforeAll(() => {
  global.dotIconProps = {
    width: 45,
    height: 45
  };
});

afterAll(() => {
  delete global.dotIconProps;
});

/* global dotIconProps */
describe('DotIcon.js', () => {
  it('matches the snapshot', () => {
    const tree = renderer
      .create(<DotIcon {...dotIconProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    const dotIcon = render(<DotIcon {...dotIconProps} />);
    expect(dotIcon).toBeDefined();
  });
});
