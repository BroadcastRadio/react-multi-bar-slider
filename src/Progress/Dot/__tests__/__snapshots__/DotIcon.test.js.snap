// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DotIcon.js matches the snapshot 1`] = `
<img
  className="css-19tq53f"
/>
`;
